pkg/hubble/observer/local_node_watcher_test.go

```diff
@@ -73,11 +77,18 @@ func Test_LocalNodeWatcher(t *testing.T) {
 		store.Update(func(ln *node.LocalNode) {
 			*ln = updatedNode
 		})
-		var flow flowpb.Flow
-		stop, err := watcher.OnDecodedFlow(ctx, &flow)
-		require.False(t, stop)
-		require.NoError(t, err)
-		require.Equal(t, updatedNodeLabelSlice, flow.GetNodeLabels())
+		require.Eventually(
+			t,
+			func() bool {
+				var flow flowpb.Flow
+				stop, err := watcher.OnDecodedFlow(ctx, &flow)
+				require.False(t, stop)
+				require.NoError(t, err)
+				return slices.Equal(updatedNodeLabelSlice, flow.GetNodeLabels())
+			},
+			time.Second,
```

Comment:
I guess will flake every now and then with "just" one second, because CI. But we can fix it with `synctest` once go 1.25 is out, so I'm not too worried. Maybe you can make it 10s.